src/Toggle/MysqlActivatedToggleGateway.php

```diff
@@ -43,20 +62,21 @@ public function __construct( Connection $connection, IsToggleActive $toggleCheck
      */
     public function getAllMyActivatedToggles( $userIdentifier )
     {
-        $activatedToggles = [ ];
-
-        $queryBuilder = new QueryBuilder( $this->connection );
-        $queryBuilder
-            ->select( '*, toggle.id as toggleId' )
-            ->from( 'toggle' );
-        $data = $queryBuilder->execute()->fetchAll();
-        $toggles = $this->getAllTogglesFromGivenSqlResult( $data );
-
-        foreach ( $toggles as $toggle ) {
-            if ( $this->toggleChecker->isToggleActive( $toggle->getName() ) ) {
-                $activatedToggles[] = $toggle;
+        $toggles = [ ];
+
+        $data = $this->connection->fetchAll(
+            'SELECT *, toggle.id as toggleId
+             FROM `toggle`
+             LEFT JOIN `toggle_marketing_information`
+             ON toggle.id = toggle_marketing_information.toggle_id
+             ORDER BY toggle.id ASC' );
+
+        foreach ( $data as $row ) {
+            if ( $this->toggleChecker->isToggleActive( $row[ 'name' ], $this->user, $this->group ) ) {
```

Comment:
I don't think he has too much choice. There's no interface for determining it for multiple toggles, the only implementation does it for one toggle at a time.

Performance issues will need to be sorted out later I guess, iterate or what everyone always says :D